src/libaktualizr/uptane/imagerepository.cc

```diff
@@ -254,7 +264,13 @@ void ImageRepository::updateMeta(INvStorage& storage, const IMetadataFetcher& fe
         fetch_snapshot = false;
         LOG_DEBUG << "Skipping Image repo Snapshot download; stored version is still current.";
       } catch (const Uptane::Exception& e) {
-        LOG_ERROR << "Image repo Snapshot verification failed: " << e.what();
+        if (!prev_timestamp.isInitialized() || prev_timestamp.snapshot_hashes() != timestamp.snapshot_hashes()) {
```

Comment:
This should work in our case. However, it won't work in all use-cases, kind of nitpicking here.

This work based on assumption that that snapshot hashes in two timestamp versions are stored in the same order.
Also, it may happen that a timestamp v1 has only sha256 hash while timestamp v2 has only sha512 hash of the same snapshot. 
It's not critical, it just means that in some edge cases (listed above) the debug message is printed instead of error one. I am even not sure how that happen. If a local snapshot was broken somehow, then the hash will mismatch and we need to re-pull it anyway so this is the "debug" case. If it wasn't broken than this is very unlikely that `snapshot = Snapshot(RepositoryType::Image(), Utils::parseJSON(snapshot_raw), std::make_shared<MetaWithKeys>(root));` will fail because this check was done previously anyway.

@detsch This is up to you, if you want to fix/improve it feel free to do it if not I am fine with change :).

Reply:
btw, I think doing that `if` in a dedicated `const Uptane::SecurityException& e)` catch can help to avoid that issue too.

Reply:
Moved the `if` a the specific `const Uptane::SecurityException& e)` catch, it makes sense.
I believe it is fine to deal with changes in the order and/or type of hashes listed in the timestamp the same way as if there were actual changes in the snapshot, in terms of log severity. Not that I have a strong opinion about it, but after giving it some thought, it seems to me that is better keep the current simplicity.

Reply:
OK,makes sense.